### ISSUE TITLE
Style/ClassAndModuleChildren cop incorrectly corrects parent `class` to `module`

### ISSUE THREAD

**Issue by lopopolo** [rubocop-hq/rubocop#9128]
RuboCop incorrectly expands compact style class reopenings to use `module` for the parent if the parent is a `Class`.

--------

## Expected behavior

Describe here how you expected RuboCop to behave in this particular situation.

### Input

```ruby
# frozen_string_literal: true

# Foo class
class Foo
  # Foo::Bar class
  class Bar
    # Foo::Bar#baz method
    def baz
      'baz'
    end
  end
end

class Foo::Bar
  # Foo::Bar#quux method
  def quux
    'quux'
  end
end
```

### Expected output

```ruby
# frozen_string_literal: true

# Foo class
class Foo
  # Foo::Bar class
  class Bar
    # Foo::Bar#baz method
    def baz
      'baz'
    end
  end
end

class Foo
  class Bar
    # Foo::Bar#quux method
    def quux
      'quux'
    end
  end
end
```

## Actual behavior

Describe here what actually happened.
Please use `rubocop --debug` when pasting rubocop output as it contains additional information.

### Input

```ruby
# frozen_string_literal: true

# Foo class
class Foo
  # Foo::Bar class
  class Bar
    # Foo::Bar#baz method
    def baz
      'baz'
    end
  end
end

class Foo::Bar
  # Foo::Bar#quux method
  def quux
    'quux'
  end
end
```

### Expected output

```ruby
# frozen_string_literal: true

# Foo class
class Foo
  # Foo::Bar class
  class Bar
    # Foo::Bar#baz method
    def baz
      'baz'
    end
  end
end

module Foo
  class Bar
    # Foo::Bar#quux method
    def quux
      'quux'
    end
  end
end
```

### Executing the bad correction code

```console
$ cat rubocop_compact_autocorrect_bad.rb
# frozen_string_literal: true

# Foo class
class Foo
  # Foo::Bar class
  class Bar
    # Foo::Bar#baz method
    def baz
      'baz'
    end
  end
end

module Foo
  class Bar
    # Foo::Bar#quux method
    def quux
      'quux'
    end
  end
end
$ ruby rubocop_compact_autocorrect_bad.rb
Traceback (most recent call last):
rubocop_compact_autocorrect_bad.rb:14:in `<main>': Foo is not a module (TypeError)
```

## Steps to reproduce the problem

This is extremely important! Providing us with a reliable way to reproduce
a problem will expedite its solution.

bundle exec rubocop -A [source]

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler.
If you see extension cop versions (e.g. `rubocop-performance`, `rubocop-rspec`, and others)
output by `rubocop -V`, include them as well. Here's an example:

```
$ bundle exec rubocop -V
1.3.1 (using Parser 2.7.2.0, rubocop-ast 1.1.1, running on ruby 2.6.6 x86_64-darwin20)
```
